sklearn/metrics/pairwise.py

```diff
@@ -426,6 +426,57 @@ def pairwise_distances_argmin(X, Y, axis=1, metric="euclidean",
                                          metric_kwargs)[0]
 
 
+def haversine_distances(X, Y=None, ):
+    """
+    Compute haversine distance between samples in X and Y.
+
+    Haversine (Spherical) Distance
+
+    The Haversine distance is the angular distance between two points on
+    the surface of a sphere. The first distance of each point is assumed
+    to be the latitude, the second is the longitude, given in radians.
+    The dimension of the points must be 2.
+
+    .. math::
+       D(x, y) = 2\arcsin[\sqrt{\sin^2((x1 - y1) / 2)
+                                + cos(x1)cos(y1)sin^2((x2 - y2) / 2)}]
+
+    Parameters
+    ----------
+    X : array_like
+        with shape (n_samples_X, 2).
+
+    Y : array_like
+        with shape (n_samples_Y, 2).
```

Comment:
shape information can go in the first line itself... More description, if available, can follow next :) Like [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/metrics/pairwise.py#L159).